tests/test_pipeline_media.py

```diff
@@ -90,6 +91,30 @@ def test_modify_media_request(self):
         self.pipe._modify_media_request(request)
         assert request.meta == {'handle_httpstatus_all': True}
 
+    def test_cache_result(self):
+        request = Request('http://url')
+        response = Response('http://url', body=b'', request=request)
+
```

Comment:
This test is also very synthetic. We should add a comment here explaining the lines below are intended to prepare a specific Failure object that holds references to Response/Request objects in its context.

The test docstring should make a reference to the memory leak it is trying to prevent.